governance/xc_admin/packages/xc_admin_frontend/components/tabs/UpdateProductMetadata.tsx

```diff
@@ -0,0 +1,346 @@
+    Object.keys(data)
+      .sort()
+      .forEach((key) => {
+        const sortedInnerData: any = {}
+        Object.keys(data[key])
+          .sort()
+          .forEach((innerKey) => {
+            sortedInnerData[innerKey] = data[key][innerKey]
+          })
+        sortedSymbolToProductMetadataMapping[key] = sortedInnerData
+      })
+
+    return sortedSymbolToProductMetadataMapping
+  }
+
+  // function to download json file
+  const handleDownloadJsonButtonClick = () => {
+    const dataStr =
+      'data:text/json;charset=utf-8,' +
+      encodeURIComponent(JSON.stringify(data, null, 2))
+    const downloadAnchor = document.createElement('a')
+    downloadAnchor.setAttribute('href', dataStr)
+    downloadAnchor.setAttribute('download', 'products.json')
+    document.body.appendChild(downloadAnchor) // required for firefox
+    downloadAnchor.click()
+    downloadAnchor.remove()
+  }
+
+  // function to upload json file and update productMetadataChanges state
+  const handleUploadJsonButtonClick = () => {
+    const uploadAnchor = document.createElement('input')
+    uploadAnchor.setAttribute('type', 'file')
+    uploadAnchor.setAttribute('accept', '.json')
+    uploadAnchor.addEventListener('change', (e) => {
+      const file = (e.target as HTMLInputElement).files![0]
+      const reader = new FileReader()
+      reader.onload = (e) => {
+        if (e.target) {
+          const fileData = e.target.result
+          if (!isValidJson(fileData as string)) return
+          const fileDataParsed = sortData(JSON.parse(fileData as string))
+          const changes: Record<string, ProductMetadataInfo> = {}
+          Object.keys(fileDataParsed).forEach((symbol) => {
+            if (
+              JSON.stringify(data[symbol]) !==
+              JSON.stringify(fileDataParsed[symbol])
+            ) {
+              changes[symbol] = {
+                prev: data[symbol],
+                new: fileDataParsed[symbol],
+              }
+            }
+          })
+          setProductMetadataChanges(changes)
+          openModal()
+        }
+      }
+      reader.readAsText(file)
+    })
+    document.body.appendChild(uploadAnchor) // required for firefox
+    uploadAnchor.click()
+    uploadAnchor.remove()
+  }
+
+  // check if uploaded json is valid json
+  const isValidJson = (json: string) => {
+    try {
+      JSON.parse(json)
+    } catch (e: any) {
+      toast.error(capitalizeFirstLetter(e.message))
+      return false
+    }
+    // check if json keys are existing products
+    const jsonParsed = JSON.parse(json)
+    const jsonSymbols = Object.keys(jsonParsed)
+    const existingSymbols = Object.keys(data)
+    // check that jsonSymbols is equal to existingSymbols no matter the order
+    if (
+      JSON.stringify(jsonSymbols.sort()) !==
+      JSON.stringify(existingSymbols.sort())
+    ) {
+      toast.error('Symbols in json file do not match existing symbols!')
```

Comment:
Can you error if there are duplicate keys in the uploaded JSON

Reply:
this is a general error that includes this situation -- do you mean do a specific check just for duplicate keys?

Reply:
do you mean json keys in the metadata?

Reply:
Duplicate symbol